v8/config/krb5conf.go

```diff
@@ -726,3 +731,100 @@ func (c *Config) JSON() (string, error) {
 	}
 	return string(b), nil
 }
+
+var expandMap map[string]string
+
+func init() {
+	expandMap = makeExpandMap()
+}
+
+func makeExpandMap() (out map[string]string) {
+	out = make(map[string]string)
+	vars := PkgConfigVars(nil)
+
+	out["TEMP"] = os.TempDir()
+
+	user, err := user.Current()
+	if err == nil {
+		out["uid"] = user.Uid
+		out["USERID"] = user.Uid
+		out["username"] = user.Username
+	}
+
+	if runtime.GOOS == "windows" {
+		out["euid"] = out["uid"]
+	} else {
+		out["euid"] = fmt.Sprintf("%d", os.Geteuid())
+	}
+
+	out["null"] = ""
+	out["LIBDIR"] = vars["libdir"]
+	out["BINDIR"] = vars["exec_prefix"] + "/bin"
+	out["SBINDIR"] = vars["exec_prefix"] + "/sbin"
+
+	return
+}
+
+func ExpandParams(in string) (out string) {
+	out = in
+
+	for k, v := range expandMap {
+		repl := fmt.Sprintf("%%{%s}", k)
+		out = strings.ReplaceAll(out, repl, v)
+	}
+	return
+}
+
+// PkgConfigVars returns useful variables from the krb5 pkg-config file, if it is found
+// otherwise returns a set of defaults
+func PkgConfigVars(cfg *string) (out map[string]string) {
+	out = make(map[string]string)
+	for k, v := range defaultPkgConfigVars {
+		out[k] = v
+	}
+
+	if cfg == nil {
+		// find krb5-config
+		file, err := exec.LookPath("krb5-config")
```

Comment:
Can you share some details on this file please? I've not been able to understand what this relates to. thanks